### ISSUE TITLE
API request

### ISSUE THREAD

**Issue by VladoCC** [progwml6/HungerOverhaul#196]
Okey, @progwml6 said that I need towrite issue here to ask about api for Get Back To Home mod.
I want to get:
-Food values (hunger, saturation)
-Some way to get how exhaustion works with your mod. You changes it in some pats, right?

**Comment by squeek502** [comment]
Food values can be retrieved using the AppleCore API (which is what Hunger Overhaul uses under the hood to modify food values):

- https://github.com/squeek502/AppleCore/blob/HEAD/java/squeek/applecore/api/IAppleCoreAccessor.java#L23-L42

Example:
```java
AppleCoreAPI.accessor.getFoodValues(itemStack);
```

HO affects exhaustion in a few ways:

- [Adds constant exhaustion increase when `Config.constantHungerLoss` is true](https://github.com/progwml6/HungerOverhaul/blob/4b77fa61296a06a8ba39133ebfb3aa7789095f53/src/main/java/iguanaman/hungeroverhaul/module/event/HungerOverhaulEventHook.java#L190-L198)
- [Disables losing hunger from exhaustion when `Config.hungerLossRatePercentage` is 0](https://github.com/progwml6/HungerOverhaul/blob/5502e5a3b2eb6db1f9ee36c1912f780949f5a82b/src/main/java/iguanaman/hungeroverhaul/module/food/FoodEventHandler.java#L60-L70)
- [Modifies the threshold for losing hunger from exhaustion based on `Config.hungerLossRatePercentage`](https://github.com/progwml6/HungerOverhaul/blob/5502e5a3b2eb6db1f9ee36c1912f780949f5a82b/src/main/java/iguanaman/hungeroverhaul/module/food/FoodEventHandler.java#L73-L91) (the modified max exhaustion can be retrieved using the AppleCore API like so: `AppleCoreAPI.accessor.getMaxExhaustion(player)`; [real world example here](https://github.com/squeek502/AppleSkin/blob/1.10.2/java/squeek/appleskin/helpers/HungerHelper.java#L14-L20))

**Comment by VladoCC** [comment]
If my mod was loaded earlier than HO, can it get values changed by HO from AppleCore API?

And is it all changes to exhaustion? I think that I saw some info about formula for speeding down exhaustion when player is low on hunger.

**Comment by squeek502** [comment]
> If my mod was loaded earlier than HO, can it get values changed by HO from AppleCore API?

Yes, load order doesn't matter at all for this (or anything else AppleCore related, I believe).

> And is it all changes to exhaustion? I think that I saw some info about formula for speeding down exhaustion when player is low on hunger.

Depends what you mean by exhaustion. If you mean the `FoodStats.foodExhaustionLevel` field, then, yes, what I linked are the only changes to how that works. If you mean 'exhaustion' as a general concept, then HO does do some other stuff like [apply potion effects when the player is low on hunger/health](https://github.com/progwml6/HungerOverhaul/blob/5502e5a3b2eb6db1f9ee36c1912f780949f5a82b/src/main/java/iguanaman/hungeroverhaul/module/event/HungerOverhaulEventHook.java#L216-L288).
